.changeset/fifty-guests-kick.md

```diff
@@ -0,0 +1,11 @@
+---
+"barnard59-s3": minor
+---
+
+Allow `getObject` and `putObject` operations to be used inside a barnard59 pipeline, as they are now returning a stream.
+They can be used at any place in the list of steps, as it will not change anything in the stream.
```

Comment:
I'm a little uneasy about this. Are those just no-op steps with side effects? I don't think we have any other step that behave like this

Can you show me an example usage in a pipeline?

Reply:
You can start a pipeline with a `getObject`, then just use that file for other steps, and it will be working.

Similarly, if you want to store data in S3, you have the n-1 th step that writes the data into a file, and you can use `putObject` to store that file into a S3 bucket.

By side effects, the only one is that it's writing a file on the file system (`getObject`) or in a S3 bucket (`putObject`). No other side effect ; if so, it's a bug that should be fixed.

Without using streams for them was, it was impossible to add those steps in any pipeline, and then we can just remove the s3 package.